packages/preferences/src/browser/util/preference-layout.ts

```diff
@@ -0,0 +1,382 @@
+                settings: ['timeline.*']
+            },
+            {
+                id: 'features.toolbar',
+                label: nls.localize('theia/preferences/toolbar', 'Toolbar'),
+                settings: ['toolbar.*']
+            },
+            {
+                id: 'features.notebook',
+                label: nls.localizeByDefault('Notebook'),
+                settings: ['notebook.*', 'interactiveWindow.*']
+            },
+            {
+                id: 'features.mergeEditor',
+                label: nls.localizeByDefault('Merge Editor'),
+                settings: ['mergeEditor.*']
+            },
+            {
+                id: 'features.chat',
+                label: nls.localizeByDefault('Chat'),
+                settings: ['chat.*', 'inlineChat.*']
+            }
+        ]
+    },
+    {
+        id: 'application',
+        label: nls.localizeByDefault('Application'),
+        children: [
+            {
+                id: 'application.http',
+                label: nls.localizeByDefault('HTTP'),
+                settings: ['http.*']
+            },
+            {
+                id: 'application.keyboard',
+                label: nls.localizeByDefault('Keyboard'),
+                settings: ['keyboard.*']
+            },
+            {
+                id: 'application.update',
+                label: nls.localizeByDefault('Update'),
+                settings: ['update.*']
+            },
+            {
+                id: 'application.telemetry',
+                label: nls.localizeByDefault('Telemetry'),
+                settings: ['telemetry.*']
+            },
+            {
+                id: 'application.settingsSync',
+                label: nls.localizeByDefault('Settings Sync'),
+                settings: ['settingsSync.*']
+            },
+            {
+                id: 'application.experimental',
+                label: nls.localizeByDefault('Experimental'),
+                settings: ['application.experimental.*']
+            },
+            {
+                id: 'application.other',
+                label: nls.localizeByDefault('Other'),
+                settings: ['application.*']
+            }
+        ]
+    },
+    {
+        id: 'security',
+        label: nls.localizeByDefault('Security'),
+        settings: ['security.*'],
+        children: [
+            {
+                id: 'security.workspace',
+                label: nls.localizeByDefault('Workspace'),
+                settings: ['security.workspace.*']
+            }
+        ]
+    },
+    {
+        id: 'extensions',
+        label: nls.localizeByDefault('Extensions'),
+        children: [
+            {
+                id: 'extensions.hosted-plugin',
+                label: nls.localize('theia/preferences/hostedPlugin', 'Hosted Plugin'),
+                settings: ['hosted-plugin.*']
+            }
+        ]
+    }
+];
+
+@injectable()
+export class PreferenceLayoutProvider {
```

Comment:
It doesn't have a state, but I think it's reasonable to assume that adopters want to override it. It's actually necessary in case adopters want to add a new preference group (something like `Compiler`, `SDK` or similar) and want to offer localization for those groups.

In that case, adopters need to add a new group to the array returned by `getLayout`. They could theoretically also just change `DEFAULT_LAYOUT` but that wouldn't be good API IMO (and pretty non-obvious).

Reply:
Fair enough.